clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -3723,27 +3723,34 @@ private Integer nodeFor(ConfigResource resource) {
 
         List<MemberIdentity> membersToRemove = new ArrayList<>();
         for (final MemberDescription member : members) {
+            MemberIdentity memberIdentity = new MemberIdentity()
+                .setReason(reason);
```

Comment:
QQ: seems `reason` is also used in `equals`, is it ok to add this? Will this cause `membersToRemove` not found from somewhere?

Reply:
Good question. Let me double check this. For the context, we were doing this before this patch as well.

Reply:
This does not seem to be an issue. I have extended the unit test to ensure that the look up works as we expect.